encoding/x/protobuf/inbound.go

```diff
@@ -78,9 +81,16 @@ func (u *unaryHandler) Handle(ctx context.Context, transportRequest *transport.R
 		}
 		responseData = protoBuffer.Bytes()
 	}
+	if isRawResponse(transportRequest.Headers) {
+		responseWriter.AddHeaders(getRawResponseHeaders())
```

Comment:
yeah, staring at this I'm still a bit confused, are we talking between the encoding and transport via headers on the request/response?